packages/@react-spectrum/dialog/src/Dialog.tsx

```diff
@@ -63,7 +63,7 @@ function Dialog(props: SpectrumDialogProps, ref: DOMRef) {
   let slots = useMemo(() => ({
     hero: {UNSAFE_className: styles['spectrum-Dialog-hero']},
     header: {UNSAFE_className: styles['spectrum-Dialog-header']},
-    heading: {UNSAFE_className: classNames(styles, 'spectrum-Dialog-heading', {'spectrum-Dialog-heading--noHeader': !hasHeader}), ...titleProps},
+    heading: {UNSAFE_className: classNames(styles, 'spectrum-Dialog-heading', {'spectrum-Dialog-heading--noHeader': !hasHeader}), level: 2, ...titleProps},
```

Comment:
if we do this, then no one can make a dialog with h3 or anything else, it'll always be h2

Reply:
otherwise probably best if we waited and handled this with part of typography

Reply:
Right now it is always h3 now anyway right? So it would be no different. Later we will probably fix this with auto headings

Reply:
Right now it's h3 by default, but users can override by setting the `level` prop. But with slots, they won't be able to override anymore (slots always win). I can't really think of a use case to make it higher than h2 though...

Reply:
I'm a bit confused by the ask here. An earlier commit added a public API for a `level` prop on the `Dialog` that would default to `2`: https://github.com/adobe/react-spectrum/pull/2633/commits/53fe561ad6ab7afcbb8f9c66a02b040960e8b72b. Is this what we want, or is it better just to explicitly set to `level={2}` on the slot as done here?

Reply:
The issue with adding it to the slot is that we use slots to enforce certain things in children. So if you put a level 2 in the slots, even if someone did `<Heading level={5}>` it'd still produce a level 2 Heading. @devongovett 's question is, would anyone need to either set a Dialog's Heading to anything other than 2?